core/frontend/src/tile/TiledGraphicsProvider.ts

```diff
@@ -18,9 +18,18 @@ import { TileTreeReference } from "./internal";
  * @extensions
  */
 export interface TiledGraphicsProvider {
-  /** For each [[TileTreeReference]] belonging to this provider that should be drawn in the specified [[Viewport]], apply the provided function. */
+  /** For each [[TileTreeReference]] belonging to this provider that should be drawn in the specified [[Viewport]], apply the provided function.
+   * This method is inefficient because it does not permit the caller to prematurely halt iteration; and awkward because `func` cannot be `async` nor
+   * return any value.
+   * Implementations should implement [[getReferences]], and callers should prefer to call [[TiledGraphicsProvider.getTileTreeRefs]].
+   */
   forEachTileTreeRef(viewport: Viewport, func: (ref: TileTreeReference) => void): void;
```

Comment:
should this be deprecated?

Reply:
Can't be. New `getReferences` is necessarily optional because making it required would be a breaking change.